### ISSUE TITLE
what are the key differences of Revideo compare to MotionCanvas?

### ISSUE THREAD

**Issue by mnlbox** [redotvideo/revideo#50]
Hi,

Thanks for the project. I'm new in video generation in general I'm trying to figure out what is the key difference of Revideo compare to MotionCanvas?

In the homepage you mentioned it's a fork version of MotionCanvas with more features but here in Github it's not appeared as a regular forked project. 🤔

Why it's a separate tools with separate brand/name and even separate repo and not a plugin or regular forked version of MotionCanvas?
Are you providing any extra service that is not available on MotionCanvas? (any cloud, saas version,...)
What are the key differences in general?
I see audio capabilities and parameterized video mentioned in the home page but it seems MotionCanvas also suppprts audio so I'm a bit confused now. 🤔

It would be nice if you can clarify this, then I can choose my tools/library for the project a bit simpler.

Sorry if I asked many questions like an interviewer 😬
And again thanks for the awesome projects. At the end more products and competitions is a fuel for this specific niche of products (React video generators)

**Comment by justusmattern27** [comment]
Hey, these are very fair questions!

We initially tried to build Revideo as a set of Motion Canvas plugins, but unfortunately, the plugin API was not flexible enough for this. We'll release a blog post soon that explains the decision behind the fork in detail, I'll let you know when it is released.

In general, Motion Canvas and Revideo have very different goals: 

- Motion Canvas aims to be a standalone editor for web animations, and not a library in the common sense that others integrate into their apps - see [this discussion](https://github.com/orgs/motion-canvas/discussions/1015) for details
- Revideo focuses on video, and aims to build a library that others can integrate into their apps. Essentially, we want to provide APIs that enable developers to build video editing apps themselves. We have also built a mini SaaS that shows how you can do this; you can find its repository [here](https://github.com/redotvideo/revideo-saas-template/tree/main)

Concretely, this translates to the following features so far:
- **Faster Video Rendering:** We have made video rendering up to 10x faster compared to Motion Canvas (see #33 )
- **Exposing Rendering APIs**: In Motion Canvas, you have to use the editor UI and press a button to render videos. We have exposed this functionality as a nodejs function call (#4 ) and a web endpoint, so that you can integrate this functionality in your apps (#29 )
- **Audio Support**: Motion Canvas has audio support, but it only allows you to add a single audio file over the duration of the video. With Revideo, you can use and edit multiple audio files and let them play at arbitrary times. Motion Canvas also doesn't export the audio stream of videos (i.e. if you use a video with sounds in Motion Canvas, it will be muted after export), whereas Revideo does (#3)

To summarize: If you want to create one-off animations with a visual editor, Motion Canvas is good for you. If you need a library that you can integrate into an app, Revideo is probably better suited for it.

I hope this helps!

**Comment by mnlbox** [comment]
Thanks @justusmattern27 for your quick reply.
Looking forward to reading the blog post that you mentioned to get more insights.

I think I should start creating some stuff with Revideo to see if it fits our needs. As I mentioned I'm new in the video/image generation topic so any type of blog posts, guides and ... can be helpful and I highly appreciate it.

**Comment by justusmattern27** [comment]
@mnlbox of course! If you want hands-on examples, feel free to check out our open source [example projects](https://docs.re.video/category/code-examples)

**Comment by justusmattern27** [comment]
@mnlbox the blog post regarding the fork is now live [here](https://re.video/blog/fork)! I hope this clears things up - will close the issue for now :)
